src/shared/sol-util-file.c

```diff
@@ -86,6 +86,10 @@ sol_util_write_file(const char *path, const char *fmt, ...)
     va_list ap;
     int ret;
 
+    ret = access(path, 0);
+    if (ret < 0)
+        return ret;
+
```

Comment:
Yes, segfault in sol_iio_read_channel_value because device is NULL which failed to creation in sol_iio_open. FYI, The segment fault appear when we set buffer_size=-1